app/services/PenaltiesService.scala

```diff
@@ -167,7 +167,7 @@ class PenaltiesService @Inject()(config: FrontendAppConfig,
                     (implicit ec: ExecutionContext, hc: HeaderCarrier): Future[Seq[PenaltySchemes]] =
     for {
       penalties <- fsConnector.getPsaFS(psaId)
-      _ <- fiCacheConnector.save(Json.obj("pstrs" -> penalties.map(_.pstr)))
+      _ <- fiCacheConnector.save(Json.obj("pstrs" -> penalties.map(_.pstr).distinct))
```

Comment:
Are we OK not filtering out the associated schemes before we save in new mongo cache? It's unnecessary saving associated schemes and there could be quite a lot of them.

Reply:
Good point. Will look at that.

Reply:
Actioned